src/css/wizard-navigation-bar.scss

```diff
@@ -140,120 +120,155 @@ $wz-param-indicator-color: null !global;
+@mixin wz-define-styles() {
+  @each $layout in $wz-layouts {
+    aw-wizard-navigation-bar.#{$layout} {
+      @each $indicator-style in $wz-indicator-styles {
+        &.#{$indicator-style} ul.steps-indicator {
+
+          @if $indicator-style == 'small' {
+            @include wz-define-style($layout, $width: $wz-small-dot-width, $height: $wz-small-dot-height, $border-width: 0) {
+              background-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'background-color');
+            }
+          }
+
+          @else if $indicator-style == 'large-filled' {
+            @include wz-define-style($layout, $border-width: 0) {
+              background-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'background-color');
+            }
+          }
+
+          @else if $indicator-style == 'large-empty' {
+            @include wz-define-style($layout) {
+              $border-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'border-color');
+              @if $border-color != null {
+                // To optimize CSS output size, do not define any border styles when border-color is null
+                border: $wz-param-indicator-border-width solid $border-color;
+              }
+            }
+          }
+
+          @else if $indicator-style == 'large-filled-symbols' {
+            @include wz-define-style($layout, $border-width: 0) {
+              background-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'background-color');
+              color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'symbol-color');
+            }
+          }
+
+          @else if $indicator-style == 'large-empty-symbols' {
+            @include wz-define-style($layout) {
+              $border-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'border-color');
+              @if $border-color != null {
+                // To optimize CSS output size, do not define any border styles when border-color is null
+                border: $wz-param-indicator-border-width solid $border-color;
+              }
+              color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'symbol-color');
+            }
+          }
+        }
+      }
     }
   }
 }
 
 
-// Helper mixin to define indicator styles for 'horizontal' and 'vertical' layouts
-@mixin wz-define-layout-styles() {
-  @each $layout in 'horizontal', 'vertical' {
-    aw-wizard-navigation-bar.#{$layout} {
-      @include wz-define-styles($layout);
+// Retrieve color from the color definitions map ($wz-colors)
+//
+// Arguments:
+//   $indicator-style: Indicator style: 'small', 'large-filled' etc.
+//   $step-state:      Wizard step state: small', 'large-filled' etc.
```

Comment:
Are you sure `$step-state` takes the same values as `$indicator-style`?

Reply:
Definitely not :-) Fixed the comment in 5a9fd8a